qp_shotgun/shogun/tests/test_shogun.py

```diff
@@ -0,0 +1,167 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import main
+from qiita_client.testing import PluginTestCase
+import os
+from os import remove
+from os.path import exists, isdir, join
+from shutil import rmtree
+from tempfile import TemporaryDirectory
+from qp_shotgun import plugin
+from qp_shotgun.shogun.utils import (
+    get_dbs, get_dbs_list, generate_shogun_dflt_params)
+from qp_shotgun.shogun.shogun import (
+    generate_shogun_align_commands, _format_params,
+    generate_shogun_assign_taxonomy_commands, generate_fna_file,
+    generate_shogun_functional_commands)
+
+SHOGUN_PARAMS = {
+    'Database': 'database', 'Aligner tool': 'aligner',
+    'Taxonomic Level': 'levels', 'Number of threads': 'threads'}
+
+
+class ShogunTests(PluginTestCase):
+    maxDiff = None
+
+    def setUp(self):
+        plugin("https://localhost:21174", 'register', 'ignored')
+
+        self.db_path = os.environ["QC_SHOGUN_DB_DP"]
+        self.params = {
+            'Database': join(self.db_path, 'shogun'),
+            'Aligner tool': 'bowtie2',
+            'Taxonomic Level': 'all',
+            'Number of threads': 1
+        }
+        self._clean_up_files = []
+
+    def tearDown(self):
+        for fp in self._clean_up_files:
+            if exists(fp):
+                if isdir(fp):
+                    rmtree(fp)
+                else:
+                    remove(fp)
+
+    def test_get_dbs(self):
+        db_path = self.db_path
+        obs = get_dbs(db_path)
+        exp = {'shogun': join(db_path, 'shogun')}
+
+        self.assertEqual(obs, exp)
+
+    def test_get_dbs_list(self):
+        db_path = self.db_path
+        obs = get_dbs_list(db_path)
+        exp = join(join('"'+db_path, 'shogun')+'"')
+
+        self.assertEqual(obs, exp)
+
+    def test_generate_shogun_dflt_params(self):
+        obs = generate_shogun_dflt_params()
+        exp = {
+            'shogun_bowtie2': {
+                'Database': join(self.db_path, 'shogun'),
+                'Aligner tool': 'bowtie2',
+                'Taxonomy Level': 'all',
+                'Number of threads': 1},
+            'shogun_utree': {
+                'Database': join(self.db_path, 'shogun'),
+                'Aligner tool': 'utree',
+                'Taxonomy Level': 'all',
+                'Number of threads': 1},
+            'shogun_burst': {
+                'Database': join(self.db_path, 'shogun'),
+                'Aligner tool': 'burst',
+                'Taxonomy Level': 'all',
+                'Number of threads': 1}}
+
+        self.assertEqual(obs, exp)
+
+    def test_generate_fna_file(self):
+        temp_path = os.environ['QC_SHOGUN_TEMP_DP']
```

Comment:
For tests you could use: `from tempfile import mkdtemp` to avoid having to set this variable.